site/content/documentation/reference/configuration.md

```diff
@@ -0,0 +1,458 @@
+
+The Direct Type inferencer is an Sail Adapter that performs entailment on the
+the class and instance inheritance hierarchy, asserting specific "shortcut"
+properties (`sesame:directType`, `sesame:directSubClassOf` and
+`sesame:directSubPropertyOf`) that allow querying for the _direct_ type or
+subclass of a resource.  Reasoning happens in a forward-chaining manner on
+update operations, and the entailed statements are sent to be persisted by the
+wrapped Sail.
+
+The `sail:sailType` value is `"openrdf:DirectTypeHierarchyInferencer"`.
+
+##### Example configuration
+
+```turtle
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#>.
+@prefix rep: <http://www.openrdf.org/config/repository#>.
+@prefix sr: <http://www.openrdf.org/config/repository/sail#>.
+@prefix sail: <http://www.openrdf.org/config/sail#>.
+@prefix ms: <http://www.openrdf.org/config/sail/memory#>.
+@prefix sb: <http://www.openrdf.org/config/sail/base#>.
+
+[] a rep:Repository ;
+   rep:repositoryID "example" ;
+   rdfs:label "Example Memory store with direct type entailment" ;
+   rep:repositoryImpl [
+      rep:repositoryType "openrdf:SailRepository" ;
+      sr:sailImpl [
+         sail:sailType "openrdf:DirectTypeHierarchyInferencer";
+         sail:delegate [
+             sail:sailType "openrdf:MemoryStore" ;
+             ...
+         ];
+      ]
+   ].
+```
+
+#### SHACL Sail
+
+The SHACL Sail is a Sail Adapter that performs transaction validation using the Shapes Constraint Language [SHACL](https://www.w3.org/TR/shacl/). More information about the use of the SHACL Sail can be found in [Validation with SHACL](/documentation/programming/shacl).
+
+The `sail:sailType` value of the SHACL Sail is `"rdf4j:ShaclSail"`. Its parameter namespace is `http://rdf4j.org/config/sail/shacl#`, commonly abbreviated to `ssc`. It takes the following configuration options:
+
+- `ssc:parallelValidation` (boolean): Enables parallel validation (optional).
+- `ssc:undefinedTargetValidatesAllSubjects` (boolean): Specifies if an undefined target in a shape leads to validating all subject (optional).
+- `ssc:logValidationPlans` (boolean): Specifies if validation plans are sent to the logging framework (optional).
+- `ssc:logValidationViolations` (boolean): Specifies if shape violations are sent to the logging framework (optional).
+- `ssc:ignoreNoShapesLoadedException` (boolean): Specifies if the "no shapes loaded" error is ignored (optional).
+- `ssc:validationEnabled` (boolean): Specifies if transaction valudation is enabled by default (optional).
+- `ssc:cacheSelectNodes` (boolean): Specifies if select nodes are cached (optional).
+- `ssc:globalLogValidationExecution` (boolean): Specifies if validation execution details are sent to the logging framework (optional).
+- `ssc:rdfsSubClassReasoning` (boolean): Enables RDF Schema Subclass entailment (optional).
+- `ssc:performanceLogging` (boolean): Enables performance logging (optional).
+- `ssc:serializableValidation` (boolean):
```

Comment:
Explanation: When combined with transactions using isolation level `SNAPSHOT` the validation guarantee is equivalent to using `SERIALIZABLE` without the performance impact.

Reply:
added